### ISSUE TITLE
TypeScript export

### ISSUE THREAD

**Issue by shuhei** [shuhei/rolling-window#5]
The type definition for TypeScript exports the `RollingWindow` class as the `default` while the CommonJS implementation exports the class itself.

`index.d.ts`:

```ts
export default RollingWindow;
```

v.s.

`index.js`:

```js
module.exports =  RollingWindow;
```

Thanks @DeTeam for reporting!

**Comment by shuhei** [comment]
The current setup seems to work if users have `"esModuleInterop": true`.

**Comment by shuhei** [comment]
```ts
declare namespace RollingWindow {
  interface RollingWindowOptions { /* ... */ }
}
declare class RollingWindow { /* ... */ }

export = RollingWindow;
```

`esModuleInterop: true`:

```ts
// Import the class and the options together
import RollingWindow, { RollingWindowOptions } from "@shuhei/rolling-window`;
```

`esModuleInterop: false`:

```ts
// Import the class
import * as RollingWindow from "@shuhei/rolling-window";

// Import the options
import { RollingWindowOptions } from "@shuhei/rolling-window";
// or get it from the namespace
RollingWindow.RollingWindowOptions
```

**Comment by shuhei** [comment]
Or we can leave the type definition as it is, and add ES module interop to the JS implementation like https://github.com/power-assert-js/power-assert/blob/35e3aea349d0aaa7bac31bccd2cdae0beca536a8/index.js#L65-L66 so that TypeScript users can use this package with the same interface regardless of `esModuleInterop`...?

**Comment by shuhei** [comment]
But `esModuleInterop` is necessary for Node.js anyway unless the new esm support is used?

**Comment by shuhei** [comment]
Now I think I understand the problem. Because this is a package for Node.js applications, we can assume that users' `tsconfig.json` has `"module": "commonjs"` (at least for now). The question is whether `esModuleInterop` is enabled or not.

For the following TypeScript code:

```ts
import RollingWindow, { RollingWindowOptions } from ".";

const options: RollingWindowOptions = {
    numChunks: 123
};
const rollingWindow: RollingWindow = new RollingWindow(options);
```

`tsc --target ES5 --module CommonJS test.ts` generates:

```js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const _1 = require(".");
const options = {
    numChunks: 123
};
const rollingWindow = new _1.default(options);
```

`tsc --target ES5 --module CommonJS --esModuleInterop test.ts` generates:

```js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const _1 = __importDefault(require("."));
const options = {
    numChunks: 123
};
const rollingWindow = new _1.default(options);
```

The difference is the `__importDefault` helper. Without it, the current implementation fails on runtime because `require("@shuhei/rolling-window").default` doesn't exist.

TypeScript's `"target": "commonjs"` generates JavaScript code that expects a `default` property for a default export regardless of `esModuleInterop`. `esModuleInterop` generates JavaScript code that wraps non-ES modules with `{ default: require("pacakge") }` so that non-ES modules work with the code that `"target": "commonjs"` generates.

If we can't expect users to have `esModuleInterop`, we can have the interop in the CommonJS implementation.

**Comment by shuhei** [comment]
@DeTeam Fixed in v0.1.5. https://github.com/shuhei/rolling-window/compare/v0.1.4...v0.1.5

Thanks for letting me know!

**Comment by DeTeam** [comment]
Just tried, all seem fine now, thanks @shuhei!
